src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -198,27 +198,24 @@ using IriOrUriExpression =
 using StrlenExpression =
     StringExpressionImpl<1, LiftStringFunction<decltype(strlen)>>;
 
-// LCASE
-[[maybe_unused]] auto lowercaseImpl =
-    [](std::optional<std::string> input) -> IdOrLiteralOrIri {
+template <auto toLowerOrToUpper>
+auto upperOrLowerCaseImpl =
+    [](std::optional<ad_utility::triple_component::Literal> input)
+    -> IdOrLiteralOrIri {
   if (!input.has_value()) {
     return Id::makeUndefined();
   } else {
-    return toLiteral(ad_utility::utf8ToLower(input.value()));
+    auto newContent = std::invoke(
```

Comment:
You don't need an `else` here, but the code after this can be outside the `if`, because you have an early return in the error case before.